src/Domain/Structure/Neighbors.hpp

```diff
@@ -60,15 +74,32 @@ class Neighbors {
 
   const std::unordered_set<IdType>& ids() const { return ids_; }
 
-  const OrientationMap<VolumeDim>& orientation() const { return orientation_; }
+  /// The orientations of the neighbors for each neighboring Block.
+  ///
+  /// \note All Elements within a Block share the same orientation.
+  const std::unordered_map<size_t, OrientationMap<VolumeDim>>& orientations()
+      const {
+    return orientations_;
+  }
+
+  /// Whether or not the block logical coordinates of the neighbors conform to
+  /// those of the host
+  bool are_conforming() const { return are_conforming_; }
+
+  /// The orientation of a particular neighbor.
+  const OrientationMap<VolumeDim>& orientation(const IdType& id) const;
 
   /// Reset the ids of the neighbors.
-  void set_ids_to(const std::unordered_set<IdType> new_ids) {
-    ids_ = std::move(new_ids);
-  }
+  ///
+  /// \note This should only be called to reset Element Neighbors after
+  /// h-refinement
+  void set_ids_to(std::unordered_set<IdType> new_ids);
 
   /// Add ids of neighbors.
-  /// Adding an existing neighbor is allowed.
+  ///
+  /// \note Adding an existing neighbor is allowed.
+  /// \note The additional ids must be from Blocks with the existing
+  /// orientations.
```

Comment:
why does this not have a not about post h-refinemnt? Naively I would've guessed this should only be called after h-refinement?

Reply:
It is also called by create_initial_element

Reply:
Ah, that's what I figured. I don't think you need to change anything, because unless someone is reviewing code I doubt they would ask this.